wandb/sdk/launch/docker.py

```diff
@@ -68,17 +64,32 @@ def generate_docker_image(project: _project_spec.Project, entry_cmd):
     return image_id[0]
 
 
-def build_docker_image(project: _project_spec.Project, repository_uri, base_image, api):
+def build_docker_image(project: _project_spec.Project, docker_repository_uri, base_image, api):
     """
     Build a docker image containing the project in `work_dir`, using the base image.
     """
+    import docker  # type: ignore
 
     image_uri = _get_docker_image_uri(
-        repository_uri=repository_uri, work_dir=project.dir
+        repository_uri=docker_repository_uri, work_dir=project.dir
+    )
+    if _is_wandb_local_uri(api.settings("base_url")):
+        _, _, port = _, _, port = api.settings("base_url").split(":")
+        base_url = "http://host.docker.internal:{}".format(port)
+    elif _is_wandb_dev_uri(api.settings("base_url")):
+        base_url = "http://host.docker.internal:9002"
+    else:
+        base_url = api.settings("base_url")
+    image_uri = _get_docker_image_uri(
+        repository_uri=docker_repository_uri, work_dir=project.dir
     )
 
-    wandb_project = project.docker_env["WANDB_PROJECT"]
-    wandb_entity = project.docker_env["WANDB_ENTITY"]
+    wandb_project = project.target_project
+    wandb_entity = project.target_entity
+    if project.name:
```

Comment:
i also changed this in the docker pr, we always have a project name

Reply:
Right. I'll drop this condition